p4src/linux_networking/es2k/README_LINUX_NETWORKING.md

```diff
@@ -0,0 +1,240 @@
+ip netns exec VM1 ifconfig <VF2 port> up
+```
+
+#### 4. Start OVS as a seperate process
+For learning overlay VM's source MAC we use legacy OVS as the control plane. OVS should be started as a seperate process.
+```
+export RUN_OVS=/tmp
+rm -rf $RUN_OVS/etc/openvswitch
+rm -rf $RUN_OVS/var/run/openvswitch 
+mkdir -p $RUN_OVS/etc/openvswitch/
+mkdir -p $RUN_OVS/var/run/openvswitch
+ 
+ovsdb-tool create $RUN_OVS/etc/openvswitch/conf.db /opt/p4/p4-cp-nws/share/openvswitch/vswitch.ovsschema 
+
+ovsdb-server $RUN_OVS/etc/openvswitch/conf.db --remote=punix:$RUN_OVS/var/run/openvswitch/db.sock --remote=db:Open_vSwitch,Open_vSwitch,manager_options --pidfile=$RUN_OVS/var/run/openvswitch/ovsdb-server.pid --unixctl=$RUN_OVS/var/run/openvswitch/ovsdb-server.ctl --detach 
+
+ovs-vswitchd --detach  --pidfile=$RUN_OVS/var/run/openvswitch/ovs-vswitchd.pid --no-chdir unix:$RUN_OVS/var/run/openvswitch/db.sock --unixctl=$RUN_OVS/var/run/openvswitch/ovs-vswitchd.ctl --mlockall --log-file=/tmp/ovs-vswitchd.log 
+
+alias ovs-vsctl="ovs-vsctl --db unix:$RUN_OVS/var/run/openvswitch/db.sock"
+ovs-vsctl set Open_vSwitch . other_config:n-revalidator-threads=1
+ovs-vsctl set Open_vSwitch . other_config:n-handler-threads=1
+
+ovs-vsctl  show
+```
+
+#### 5. Create VLAN representattors
+For each VM that is spawned for overlay network we need to have a port representer. We create VLAN netdev's on top of the IPDF netdev which is assigned to VSI group 3 in step-2 mentioned above.
+
+```
+ip link add link <VSI 8> name vlan1 type vlan id 1
+ip link add link <VSI 8> name vlan2 type vlan id 2
+ifconfig vlan1 up
+ifconfig vlan2 up
+```
+Note: Here the assumption is, we have created 2 overlay VM's and creating 2 port representers for those VM's.
+port representer should always bein format: `lower case string 'vlan'+'vlanID'`
+
+#### 6. Start and run ovs-vswitchd server and ovsdb-server
+Create OVS bridge, VxLAN tunnel and assign ports to the bridge.
+```
+ovs-vsctl add-br br-int
+ifconfig br-int up
+
+ovs-vsctl add-port br-int vlan1
+ovs-vsctl add-port br-int vlan2
+ifconfig vlan1 up
+ifconfig vlan2 up
+
+ovs-vsctl add-port br-int vxlan1 -- set interface vxlan1 type=vxlan options:local_ip=40.1.1.1 options:remote_ip=40.1.1.2 options:dst_port=4789
+```
+Note: Here we are creating VxLAN tunnel with VNI 0, user can create any VNI for tunneling.
+
+#### 7. Configure rules for overlay control packets.
+Configure rules to send overlay control pakcets from VM's to its respective port representers.
+
+Below configurations assumes
+- Overlay VF1 has a VSI value 14
+- Overlay VF2 has a VSI value 15
+
+These VSI values can be checked with `/usr/bin/cli_client -q -c` command on IMC. This command provides VSI ID, Vport ID and corresponding MAC addresses for all
+- IDPF netdev's on ACC
+- VF's on HOST
+- IDPF netdev's on HOST (if IDPF driver loaded by user on HOST)
+- Netdev's on IMC
+
+```
+# Configuration for control packets from overlay VF with VSI-14 to VSI-8 by tagging a VLAN ID 1

```

Comment:
Suggestion: Rules for control packets coming from overlay VF(VSI-14), IPU will add  a VLAN tag 1 and send to HOST1(VSI-8)?